### ISSUE TITLE
Document which functions only generate unrecoverable errors

### ISSUE THREAD

**Issue by slimsag** [glfw/glfw#361]
Hi, I am part of the team working on [Go bindings to GLFW](https://github.com/go-gl/glfw3). We must differentiate between errors that were caused by a programmer fault, or caused by some system fault.
### Background

Programmer fault examples, in each of these a Go runtime `panic` would occur instead of the function returning a `error`:
- `GLFW_NOT_INITIALIZED`
- `GLFW_NO_CURRENT_CONTEXT`
- `GLFW_INVALID_ENUM`
- `GLFW_INVALID_VALUE`
- `GLFW_OUT_OF_MEMORY`

System fault errors, in each of these a Go error would be returned from the called function (the programmer should expect to handle this situation):
- `GLFW_API_UNAVAILABLE`
- `GLFW_VERSION_UNAVAILABLE`
- `GLFW_PLATFORM_ERROR`
- `GLFW_FORMAT_UNAVAILABLE`

In Go an `error` is returned (and thus part of the function signature), but a `panic` is like a C++/Java exception (and not part of the function signature). Because of the difference in the function signature  **it's not enough to differentiate between these errors at runtime**.
### TL;DR;

Could documentation be added about which functions can generate which errors (e.g. can `glfwGetKey` generate errors for programmer faults or also for system faults)?

Note: it's not absolutely dire that this happen (e.g. if it's very difficult), it would really make the API more clean though.

**Comment by shurcooL** [comment]
I'm revisiting this issue, any updates on that question? Also, can you give a more concrete definition of what you mean "unrecoverable"? Unrecoverable by who? Thanks!

**Comment by elmindreda** [comment]
This doesn't really resolve the issue, but f51cf8114821b9c25d28b5572469de3c18ef446a added error lists to the reference documentation of every function.

**Comment by elmindreda** [comment]
The documentation still uses the weasel-words "possible errors include ..." but the error sections in the reference documentation should be fairly accurate now.

**Comment by elmindreda** [comment]
Is there anything else you need for this to be resolved?

**Comment by shurcooL** [comment]
> but the error sections in the reference documentation should be fairly accurate now.

Awesome, thanks for doing that!

I think that's all we need for the original issue. @slimsag, can you please verify I'm not overlooking something here?
